pkgs/by-name/ne/nexusmods-app/package.nix

```diff
@@ -1,114 +1,167 @@
     fetchSubmodules = true;
-    hash = "sha256-FzQphMhiC1g+6qmk/R1v4rq2ldy35NcaWm0RR1UlwLA=";
+    hash = "sha256-vy7gc/pS29gphkWM/KezZxXDVsD5DV02b/72pPh2Y2c=";
   };
 
-  projectFile = "NexusMods.App.sln";
+  # If the whole solution is published, there seems to be a race condition where
+  # it will sometimes publish the wrong version of a dependent assembly, for
+  # example: Microsoft.Extensions.Hosting.dll 6.0.0 instead of 8.0.0.
+  # https://learn.microsoft.com/en-us/dotnet/core/compatibility/sdk/7.0/solution-level-output-no-longer-valid
+  # TODO: do something about this in buildDotnetModule
+  projectFile = "src/NexusMods.App/NexusMods.App.csproj";
+  testProjectFile = "NexusMods.App.sln";
 
-  nativeBuildInputs = [
-    copyDesktopItems
-  ];
+  nativeBuildInputs = [ copyDesktopItems ];
 
   nugetDeps = ./deps.nix;
 
   dotnet-sdk = dotnetCorePackages.sdk_8_0;
   dotnet-runtime = dotnetCorePackages.runtime_8_0;
 
+  dotnetBuildFlags = [
+    # From https://github.com/Nexus-Mods/NexusMods.App/blob/v0.5.3/src/NexusMods.App/app.pupnet.conf#L38
+    "--property:Version=${version}"
+    "--property:TieredCompilation=true"
+
+    # From https://nexus-mods.github.io/NexusMods.App/developers/Contributing/#for-package-maintainers
+    # TODO: add DefineConstants support to buildDotnetModule
+    (
+      let
+        constants = [
+          # Tell the app it is a distro package; affects wording in update prompts
+          "INSTALLATION_METHOD_PACKAGE_MANAGER"
+
+          # Don't include upstream's 7zz binary; we use the nixpkgs version
+          "NEXUSMODS_APP_USE_SYSTEM_EXTRACTOR"
+        ];
+      in
+      "--property:DefineConstants=${lib.strings.concatStringsSep "%3B" constants}"
+    )
+  ];
+
   preConfigure = ''
-    substituteInPlace Directory.Build.props \
-      --replace '</PropertyGroup>' '<ErrorOnDuplicatePublishOutputFiles>false</ErrorOnDuplicatePublishOutputFiles></PropertyGroup>'
+    # TODO: Should we patch the desktop file here?
   '';
 
   postPatch = ''
-    ln --force --symbolic "${lib.getExe _7zzWithOptionalUnfreeRarSupport}" src/ArchiveManagement/NexusMods.FileExtractor/runtimes/linux-x64/native/7zz
+    # TODO: Should we patch the desktop file here?
   '';
 
   makeWrapperArgs = [
-    "--prefix PATH : ${lib.makeBinPath [desktop-file-utils]}"
-    "--set APPIMAGE $out/bin/${meta.mainProgram}" # Make associating with nxm links work on Linux
+    # Make associating with nxm links work on Linux
+    # TODO: is this still needed when we use INSTALLATION_METHOD_PACKAGE_MANAGER?
+    "--set APPIMAGE ${placeholder "out"}/bin/${meta.mainProgram}"
   ];
 
+  propagatedBuildInputs = [ (_7zz.override { inherit enableUnfree; }) ];
+
   runtimeDeps = [
+    # TODO: is this still needed when we use INSTALLATION_METHOD_PACKAGE_MANAGER?
+    desktop-file-utils
     fontconfig
     libICE
     libSM
     libX11
   ];
 
-  executables = [
-    nexusmods-app.meta.mainProgram
-  ];
+  executables = [ meta.mainProgram ];
```

Comment:
same here

Reply:
Thanks for the feedback! Could you clarify why though?

Reply:
We can deal with this in #336221